doschema/jsvalidation.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of DoSchema.
+# Copyright (C) 2016 CERN.
+#
+# DoSchema is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# DoSchema is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoSchema; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Merger module."""
+
+
+import collections
+
+import jsonschema
+
+from doschema.errors import JSONSchemaCompatibilityException
+
+
+class ResolverFactory(object):
+    """Factory class for resolvers."""
+
+    default_resolvers = {
+       'jsonschema': jsonschema.RefResolver
+    }
+
+    def __new__(cls, resolver):
+        """"New override for factory."""
+        if isinstance(resolver, str):
+            return ResolverFactory.default_resolvers[resolver]
+
+
+class JSONSchemaValidation(object):
+    """Class for checking compatibility between schemas."""
+
+    def __init__(self, ignore_index=True):
+        """Constructor.
+
+        :param resolver: resolver used to retrieve referenced schemas.
+        :param ignore_index: If set to True, which is default, it will ignore
+                             array indexes.
+        """
+        self.ignore_index = ignore_index
+        self.fields_types_dict = {}
+        self.schema_name = None
+
+    def start(self, schema, schema_name, resolver=False):
```

Comment:
No need for the resolver argument, it should be created by the same factory for each schema.